framework/console/widgets/Table.php

```diff
@@ -241,36 +241,37 @@ protected function renderRow(array $row, $spanLeft, $spanMiddle, $spanRight)
         $buffer = '';
         $arrayPointer = [];
         $finalChunk = [];
+        $alreadyPrintedCells = [];
         for ($i = 0, ($max = $this->calculateRowHeight($row)) ?: $max = 1; $i < $max; $i++) {
             $buffer .= $spanLeft . ' ';
             foreach ($size as $index => $cellSize) {
                 $cell = isset($row[$index]) ? $row[$index] : null;
                 $prefix = '';
+                $chunk = '';
                 if ($index !== 0) {
                     $buffer .= $spanMiddle . ' ';
                 }
                 if (is_array($cell)) {
                     if (empty($finalChunk[$index])) {
                         $finalChunk[$index] = '';
-                        $start = 0;
                         $prefix = $this->listPrefix;
                         if (!isset($arrayPointer[$index])) {
                             $arrayPointer[$index] = 0;
                         }
-                    } else {
-                        $start = mb_strwidth($finalChunk[$index], Yii::$app->charset);
                     }
-                    $chunk = mb_substr($cell[$arrayPointer[$index]], $start, $cellSize - 4, Yii::$app->charset);
+                    $chunk = $cell[$arrayPointer[$index]];
                     $finalChunk[$index] .= $chunk;
                     if (isset($cell[$arrayPointer[$index] + 1]) && $finalChunk[$index] === $cell[$arrayPointer[$index]]) {
                         $arrayPointer[$index]++;
                         $finalChunk[$index] = '';
                     }
                 } else {
-                    $chunk = mb_substr($cell, ($cellSize * $i) - ($i * 2), $cellSize - 2, Yii::$app->charset);
+                    if(!isset($alreadyPrintedCells[$index]))
+                        $chunk = $cell;
+                    $alreadyPrintedCells[$index] = true;
                 }
                 $chunk = $prefix . $chunk;
-                $repeat = $cellSize - mb_strwidth($chunk, Yii::$app->charset) - 1;
+                $repeat = $cellSize - mb_strwidth($this->escapeCharacters($chunk), Yii::$app->charset) - 1;
```

Comment:
I am not sure about the necessity of `Yii::$app->charset`, but I think it is preferred to either do
`$repeat = $cellSize - mb_strwidth(Console::stripAnsiFormat($chunk), Yii::$app->charset) - 1;`
or
`$repeat = $cellSize - Console::ansiStrlen($chunk) - 1;`
and get rid of the function below altogether as it serves no additional purpose. (after changing all instance of such code of course, I will not mark them all)

Reply:
I'm not 100% sure about `Console::ansiStrlen()` method, because it uses `mb_strlen()` inside (not `mb_strwidth()` which is used all across `Table` widget for some reason)
What about my own `escapeCharacters()` method, well initially i thought about stripping only such characters, that may have effect on string width, but since we used `Console::stripAnsiFormat()`, this probably doesn't make sense. Will do

Reply:
Agreed on `Console::ansiStrlen()`, I failed to see the usage of `mb_strwidth` instead of `mb_strlen` :unamused:

Perhaps it is a good idea to introduce a `Console::ansiStrwidth`. What is your idea about that @samdark ?

Reply:
```
    /**
     * Returns the width of the string without ANSI color codes.
     * @param string $string the string to measure
     * @return int the width of the string not counting ANSI format characters
     * @since 2.0.36
     */
    public static function ansiStrwidth($string)
    {
        return mb_strwidth(static::stripAnsiFormat($string), Yii::$app->charset);
    }
```

Reply:
I'm fine with introducing it.